src/main/java/org/radarcns/management/service/mapper/ClientDetailsMapperImpl.java

```diff
@@ -43,31 +52,41 @@ public ClientDetails clientDetailsDTOToClientDetails(ClientDetailsDTO detailsDTO
         BaseClientDetails result = new BaseClientDetails();
         result.setClientId(detailsDTO.getClientId());
         result.setClientSecret(detailsDTO.getClientSecret());
-        result.setScope(Collections.unmodifiableSet(detailsDTO.getScope()));
-        result.setResourceIds(Collections.unmodifiableSet(detailsDTO.getResourceIds()));
-        result.setAuthorizedGrantTypes(Collections.unmodifiableSet(detailsDTO.getAuthorizedGrantTypes()));
-        result.setAccessTokenValiditySeconds(detailsDTO.getAccessTokenValidity().intValue());
-        result.setRefreshTokenValiditySeconds(detailsDTO.getRefreshTokenValidity().intValue());
-        Set<GrantedAuthority> authorities = detailsDTO.getAuthorities().stream()
-            .map(SimpleGrantedAuthority::new)
-            .collect(Collectors.toSet());
-        result.setAuthorities(Collections.unmodifiableSet(authorities));
+        result.setScope(detailsDTO.getScope());
+        result.setResourceIds(detailsDTO.getResourceIds());
+        result.setAuthorizedGrantTypes(detailsDTO.getAuthorizedGrantTypes());
+        if (Objects.nonNull(detailsDTO.getAccessTokenValidity())) {
+            result.setAccessTokenValiditySeconds(detailsDTO.getAccessTokenValidity().intValue());
+        }
+        if (Objects.nonNull(detailsDTO.getRefreshTokenValidity())) {
+            result.setRefreshTokenValiditySeconds(detailsDTO.getRefreshTokenValidity().intValue());
+        }
+        if (Objects.nonNull(detailsDTO.getAuthorities())) {
+            Set<GrantedAuthority> authorities =
+                detailsDTO.getAuthorities().stream().map(SimpleGrantedAuthority::new).collect(Collectors.toSet());
+            result.setAuthorities(authorities);
+        }
         if (detailsDTO.getAutoApprove()) {
-            result.setAutoApproveScopes(Collections.unmodifiableSet(detailsDTO.getScope()));
+            result.setAutoApproveScopes(detailsDTO.getScope());
         }
-        result.setAdditionalInformation(Collections.unmodifiableMap(detailsDTO
-            .getAdditionalInformation()));
+        result.setAdditionalInformation(detailsDTO.getAdditionalInformation());
         return result;
     }
 
     @Override
     public List<ClientDetailsDTO> clientDetailsToClientDetailsDTO(List<ClientDetails> detailsList) {
+        if (Objects.isNull(detailsList)) {
+            return null;
```

Comment:
It's better to return Collections.emptyList() here, otherwise you would have to check for null everywhere it's used.